File: ftests/mpi_basic.f90
program fbasic
include 'mpif.h'
integer rank, size, ierror 

call MPI_INIT(ierror)
call MPI_COMM_SIZE(MPI_COMM_WORLD, size, ierror)
call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierror)
print*, "Hello, I'm ", rank, ' of ', size
call MPI_FINALIZE(ierror)
end

File: tests/pmpi/mpi_basic.c
/*******************************************************************************
 * Copyright 2019 UChicago Argonne, LLC.
 * (c.f. AUTHORS, LICENSE)
 *
 * This file is part of the nrm-extra project.
 * For more info, see https://github.com/anlsys/nrm-extra
 *
 * SPDX-License-Identifier: BSD-3-Clause
 *******************************************************************************/
#include <mpi.h>
#include <stdio.h>

int main(int argc, char **argv) {
  int size, rank;
  MPI_Init(&argc, &argv);
  MPI_Comm_size(MPI_COMM_WORLD, &size);
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
  fprintf(stdout, "Hello, I'm %u of %u\n", rank, size);
  MPI_Finalize();
  return 0;
}
